test/xpu_api/utilities/meta/meta.unary/meta.unary.prop/has_virtual_destructor.pass.cpp

```diff
@@ -0,0 +1,91 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+#include "support/utils_invoke.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+template <class T>
+void
+test_has_not_virtual_destructor(sycl::queue& deviceQueue)
+{
+    deviceQueue.submit([&](sycl::handler& cgh) {
+        cgh.single_task<T>([=]() {
+            static_assert(!dpl::has_virtual_destructor<T>::value);
+            static_assert(!dpl::has_virtual_destructor<const T>::value);
+            static_assert(!dpl::has_virtual_destructor<volatile T>::value);
+            static_assert(!dpl::has_virtual_destructor<const volatile T>::value);
+
+            static_assert(!dpl::has_virtual_destructor_v<T>);
+            static_assert(!dpl::has_virtual_destructor_v<const T>);
+            static_assert(!dpl::has_virtual_destructor_v<volatile T>);
+            static_assert(!dpl::has_virtual_destructor_v<const volatile T>);
```

Comment:
Why we have only "negative" test cases in this test? Should we add a test with some structure with dtor marked as `virtual`?

Reply:
Fixed